service/framework.go

```diff
@@ -842,6 +845,23 @@ func migrateTPRsToCRDs(logger micrologger.Logger, clientSet *versioned.Clientset
 			} else if err != nil {
 				logger.Log("error", fmt.Sprintf("%#v", err))
 				return
+			} else if res.Spec.Cluster.Kubernetes.CloudProvider == "" {
+				// CRO existed with empty CloudProvider, refresh
+				patch := []v1alpha1.AWSConfig{}
+				patch[0].Spec.Cluster.Kubernetes.CloudProvider = awsCloudProvider
```

Comment:
I don't follow why we work with a list here when we actually only have one custom object per iteration. Is this some specialty of the PATCH magic?

Reply:
With `type.JSONPatchType` we need an array, `type.MergePatchType` accepts the JSON format of the required change directly (https://tools.ietf.org/html/rfc7386), which feels more natural, PTAL